### PR TITLE
Portals C++ Solution

### PR DESCRIPTION
**Pull request by maggie-j-liu**

### CODE REVIEW

**Review by SansPapyrus683 — APPROVED** [review]
nothing too egregious

File: solutions/usaco-1138.mdx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	}
+}
+
+int find(int a)
+{
+	if (a == parent[a])
+	{
+		return a;
+	}
+	return parent[a] = find(parent[a]);
```
Comment (SansPapyrus683) [inline solutions/usaco-1138.mdx:46]:
```suggestion
	return a == parent[a] ? a : (parent[a] = find(parent[a]));
```

Comment (maggie-j-liu) [reply solutions/usaco-1138.mdx:46]: ehh this looks concise but i'm pretty sure it decreases readability. what do other people think?

Comment (bqi343) [reply solutions/usaco-1138.mdx:46]: I think it's fine as is. If you really want to be concise, you should use one array for parents and sizes like I do here:

https://github.com/bqi343/USACO/blob/4aa96cd195a770c3a7f8977441020036d84b4f24/Implementations/content/graphs%20(12)/DSU/DSU%20(7.6).h

Comment (SansPapyrus683) [reply solutions/usaco-1138.mdx:46]: removes 4 lines, and i don't think any people in gold don't know about the ternary expression

Comment (maggie-j-liu) [reply solutions/usaco-1138.mdx:46]: i still think the original is more readable...

**Pull request merged**